benchmarks/Benchmarks.Trace/GraphQLBenchmark.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Datadog.Trace;
+using Datadog.Trace.ClrProfiler.Emit;
+using Datadog.Trace.ClrProfiler.Integrations;
+using Datadog.Trace.Configuration;
+using GraphQL;
+using GraphQL.Execution;
+
+namespace Benchmarks.Trace
+{
+    [MemoryDiagnoser]
+    public class GraphQLBenchmark
+    {
+        private static readonly Task<ExecutionResult> Result = Task.FromResult(new ExecutionResult { Value = 42 });
+        private static readonly int MdToken;
+        private static readonly IntPtr GuidPtr;
+        private static readonly GraphQLClient Client = new GraphQLClient();
+        private static readonly ExecutionContext Context = new ExecutionContext();
+
+        static GraphQLBenchmark()
+        {
+            var settings = new TracerSettings
+            {
+                StartupDiagnosticLogEnabled = false
+            };
+
+            Tracer.Instance = new Tracer(settings, new DummyAgentWriter(), null, null, null);
+
+            var methodInfo = typeof(GraphQLClient).GetMethod("ExecuteAsync", System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Instance);
+
+            MdToken = methodInfo.MetadataToken;
+            var guid = typeof(GraphQLClient).Module.ModuleVersionId;
+
+            GuidPtr = Marshal.AllocHGlobal(Marshal.SizeOf(guid));
+
+            Marshal.StructureToPtr(guid, GuidPtr, false);
+
+            new GraphQLBenchmark().ExecuteAsync();
+
+        }
+
+        [Benchmark]
+        public int ExecuteAsync()
+        {
+            var task = (Task<ExecutionResult>)GraphQLIntegration.ExecuteAsync(Client, Context, (int)OpCodeValue.Callvirt, MdToken, (long)GuidPtr);
+
+            return task.GetAwaiter().GetResult().Value;
```

Comment:
just for curiosity, can't we add the Task<ExecutionResult> to the benchmark return type and await here?

Reply:
We probably can, but since I'm not 100% sure it's supported I feel safer that way 😅